src/main/java/no/uib/inf112/core/player/Robot.java

```diff
@@ -112,18 +110,17 @@ public void move(@NotNull Movement movement) {
     /**
      * Move the robot with given delta to new coordinates
      */
-    private void move(int deltaX, int deltaY) {
+    private void move(int deltaX, int deltaY) throws OutSideBoardException {
         x += deltaX;
         y += deltaY;
-        if (RoboRally.getCurrentMap().isOutsideBoard(x, y)) {
-            RoboRally.getPlayerHandler().mainPlayer().kill();
-            return;
+        if (GameGraphics.getRoboRally().getCurrentMap().isOutsideBoard(x, y)) {
+            throw new OutSideBoardException();
         }
         update();
     }
 
     public void teleport(int x, int y) {
-        if (RoboRally.getCurrentMap().isOutsideBoard(x, y)) {
+        if (GameGraphics.getRoboRally().getCurrentMap().isOutsideBoard(x, y)) {
             throw new IllegalArgumentException("Cannot teleport outside the map bounds. Tried to teleport to (" + x + ", " + y + ")");
```

Comment:
should it? its a difference saying that you're not allowed to do this, and that this happens

Reply:
By this logic we do not need a OutsideBoardException, as if it is outside the board in the move method above it should too throw an IllegalArgumentException.

Reply:
but by doing it this way, we do not have to surround teleport with a try catch, you are allowed to move outside the map but you are not allowed to teleport outside of the map, of course this might just be unnecessary complications but i believe it serves a purpose doing it this way

Reply:
Okay I see your point. Is it then really necessary to create a new exception just for this one case? Wouldn't it be better for the move method to return a boolean.

Reply:
It might not be, the advantage is that we force handling of the event, but since we probably aren't going to handle it a lot we could just return boolean.